package.json

```diff
@@ -1,5 +1,5 @@
 {
-  "name": "near-abi-client",
+  "name": "near-abi-client-js",
```

Comment:
Surprising that `near-api-js` uses that full qualifier when published in a registry that's exclusively for that language.

When on GitHub, it's a useful discriminant, but when on npm, it's redundant.

If anything, we should pick the first class names if they're available, or worst-case, use an organizational scope `@near/abi-client`.

- `near-api-js` should be `near-api` on NPM (seems to be taken, not sure by whom)
- `near-sdk-js` should also be `near-sdk` on NPM (taken by us)
- `near-abi-client-js` should be `near-abi-client`

Not sure what discussions took place before those package were published under those names, or if it was an oversight and is now depended on so much that it's hard to change.

Even `npm init` strips the suffix:

```console
$ mkdir near-api-js
$ cd near-api-js
$ npm init -y
$ jq .name package.json
"near-api"
```

Reply:
I fully agree with you, but this is what our other libraries do and I do not want to create an inconsistency without everyone else switching.

> a registry that's exclusively for that language.

Just noting that this is not completely true, you can find *-ts, *-as libraries among other JS-derived languages.

Reply:
Another reason is: leaving the first class name can be a vector of attack.

 - `near-sdk`, published under the name `near-sdk-js` has 610 weekly downloads.
 - while the placeholder, taken by us, under the name `near-sdk` has 69 weekly downloads, 10% of the downloads of the real package, despite being empty.

`near-api` on the other hand, has 93 weekly downloads, despite the name only being taken 3 days ago.

A potential attacker could take advantage of the fact that people's instinct leads them to the first class name and publish malicious code in that package, while depending on and exporting everything from `near-api-js` so downstream users don't notice.

Reply:
@MaximusHaximus do you have an intuition for if our lib should be published as `near-abi-client` or `near-abi-client-js`?